sql/sql_select.cc

```diff
@@ -27865,6 +27865,8 @@ void st_select_lex::print(THD *thd, String *str, enum_query_type query_type)
       }
     }
     str->append(" */ ");
+    if (join && join->cleaned)
+      return;
```

Comment:
This looks a little bit cryptic, considering the same logic below.

Please add a comment saying something like:
```
For selects with join->cleaned==true the code below will only print the 
select number. We've already printed it, so exit.
 ```